docs/decisions/0029_slug_reservations.rst

```diff
@@ -0,0 +1,114 @@
+ADR-001: Enterprise Slug Reservation System Architecture
+********************************************************
+
+:Date: June 2025
+:Authors: iloveagent57, Titans Engineering Team
+:Feature: Self-Service Purchasing - Slug Reservation Race Condition
+
+Status
+======
+**Accepted** (June 2025)
+
+Context
+=======
+
+The self-service Teams subscription checkout flow has an inherent race condition during the delay
+between Stripe checkout session creation and actual EnterpriseCustomer provisioning. This delay window consists of:
+
+1. User submits checkout form
+2. Stripe payment processing and webhook delivery
+3. Salesforce automation triggers
+4. Provisioning API call creates EnterpriseCustomer
+
+This creates opportunities for multiple users to reserve the same enterprise slug,
+leading to downstream provisioning failures in several scenarios:
+
+* Eager administrators double-clicking submit buttons
+* Frontend JavaScript errors causing duplicate API calls  
+* Malicious actors intentionally spamming checkout forms with identical slugs
```

Comment:
I'd move "multiple users trying to reserve the same slug" into the list as an additional bullet point.
```suggestion
This creates opportunities for the same enterprise slug to be provisioned twice,
leading to downstream provisioning failures in several scenarios:

* Multiple admins attempting to submit the form using the same slug.
* Eager admins double-clicking submit buttons.
* Frontend JavaScript errors causing duplicate API calls.
* Malicious actors intentionally spamming checkout forms with identical slugs.
```